frontend/src/metabase-lib/lib/metadata/Field.ts

```diff
@@ -42,9 +43,19 @@ import Base from "./Base";
  */
 
 class FieldInner extends Base {
+  id: number | FieldRef;
```

Comment:
There are a few corner cases where the `id` of a Field can be a field ref. This sucks, but we can resolve some of the pain by using the `getId` method.